pkg/operator/edgefs/s3/s3.go

```diff
@@ -138,6 +143,16 @@ func (c *S3Controller) CreateOrUpdate(s edgefsv1beta1.S3, update bool, ownerRefs
 
 func (c *S3Controller) makeS3Service(name, svcname, namespace string, s3Spec edgefsv1beta1.S3Spec) *v1.Service {
 	labels := getLabels(name, svcname, namespace)
+	httpPort := v1.ServicePort{Name: "port", Port: int32(s3Spec.Port), Protocol: v1.ProtocolTCP}
+	httpsPort := v1.ServicePort{Name: "secure-port", Port: int32(s3Spec.SecurePort), Protocol: v1.ProtocolTCP}
+
+	if s3Spec.ExternalPort != 0 {
```

Comment:
No, by specification NodePort cannot be used for a ServiceType of 'ClusterIP'.  The K8s apiserver will reject this if it is invalid
```
2019-07-25 13:33:28.254349 E | edgefs-op-s3: failed to create s3 s3-cola. failed to create s3 service. Service "rook-edgefs-s3-s3-cola" is invalid: [spec.ports[1].nodePort: Forbidden: may not be used when `type` is 'ClusterIP', spec.ports[2].nodePort: Forbidden: may not be used when `type` is 'ClusterIP']
```
Same comments as above.  If this should be validated by the operator instead of the K8s API, I can add those checks.

Reply:
Added port check to the validateService() function